Modify findc to search for more fields

qz1004 opened the pull request:
Also added the status field to GUI to make it easier to search/edit the fields. 

Closes #89

howenc commented on src/main/java/seedu/address/logic/parser/FindCommandParser.java:
```diff
@@ -19,15 +29,28 @@ public class FindCommandParser implements Parser<FindCommand> {
+        String[] statusKeyWords = argMultimap.getValue(PREFIX_STATUS).orElse("").split("\\s+");
+        String[] tagKeyWords = argMultimap.getValue(PREFIX_TAG).orElse("").split("\\s+");
+
+        GeneralPersonPredicate generalPersonPredicate = new GeneralPersonPredicate(
```
Yo, I think that maybe we should just give the arguments as the String[] instead of creating the predicate classes here.
This is to reduce the coupling so that only GeneralPersonPredicate has a dependency on the rest of Model, leaving the parser only dependent on the GeneralPersonPredicate.

I changed mine in the most my most recent PR if you want to see.

qz1004 replied:
hm okay i will make the changes

howenc reviewed (changes requested):
Changes are needed for LastContactedTime Field in Person

qz1004 commented:
Added search for LastContactedTime field and removed the predicate classes from FindCommandParser.

Also changed LastContactedTime field to optional (defaulted to LocalDateTime.MIN if left blank). 
GUI tweaked to reflect this by displaying "Not contacted yet" by default.
